### ISSUE TITLE
Should the service be aware of the implementation of the repository?

### ISSUE THREAD

**Issue by Terminator637** [shijuvar/gokit-examples#1]
I think that the service should not know or guess that the repository is a SQL database.
https://github.com/shijuvar/gokit-examples/blob/0f59bdba162d50c4f1c31cb3c6e98a7e86587171/services/order/implementation/service.go#L51

**Comment by architgupta13** [comment]
Good point. I think it could've been better wrapped in a custom error handler to help distinguish repository errors.
